sympy/ntheory/factor_.py

```diff
@@ -1381,8 +1385,41 @@ def factorint(n, limit=None, use_trial=True, use_rho=True, use_pm1=True,
             if verbose:
                 print(complete_msg)
             return factors
-
+        #Use subexponential algorithms if use_ecm
+        #Use pollard algorithms for finding small factors for 3 iterations
+        #if after small factors the number of digits of n is >= 20 then use ecm
+        iteration += 1
+        if use_ecm and iteration >= 3 and len(str(n)) >= 25:
+            break
         low, high = high, high*2
+    random.seed(1234)
```

Comment:
@jksuom @Yathartha22  I have taken here seed = 1234... but maybe we should not take this.

Reply:
I think that it is not necessary to set a seed. It may be useful to fix it when the code is being developed but the final code should work with any seed.

Reply:
Where is `random` defined here? If this is the module, it will break the random seeding for anyone else using the random module whenever this function is used. Any random number generation in SymPy. should use its own Random instance so that they don't affect each other.

Reply:
I  have removed random.seed(1234) and added a private class for random generator in `ecm` and `qs`. But here i am using `ecm_one_factor` which will require a private random generator as its parameter. What seed should i use here. maybe `high_` can be used similar to `pollard_rho` methods.

Reply:
for now i have used no seed for `ecm_one_factor`